apiserver/facades/agent/provisioner/provisioninginfo.go

```diff
@@ -113,18 +113,22 @@ func (api *ProvisionerAPI) getProvisioningInfoBase(
 	env environs.Environ,
 	endpointBindings map[string]string,
 ) (params.ProvisioningInfo, error) {
+	modelConfig, err := api.modelConfigService.ModelConfig(ctx)
```

Comment:
issue: Rather than calling ModelConfig() from the service 3 times as part of the ProvisioningInfo function, call once and pass the data.

Reply:
Done, same with model info.